concrete/src/Http/Middleware/OAuthAuthenticationMiddleware.php

```diff
@@ -35,11 +37,14 @@ class OAuthAuthenticationMiddleware implements MiddlewareInterface
     public function __construct(
         ResourceServer $oauth,
         GuzzleFactory $psrFactory,
+        UserInfoRepository $userRepository,
         HttpFoundationFactory $foundationFactory
     ) {
         $this->oauth = $oauth;
         $this->psrFactory = $psrFactory;
         $this->foundationFactory = $foundationFactory;
+        $this->userRepository = $userRepository;
+        $this->concreteRequest = \Concrete\Core\Http\Request::getInstance();
```

Comment:
This is the same as what is passed into `->process`, I don't think this is needed. It's also worth noting that this makes it impossible to unit test this class because we'd have to also be testing the `Request` dependency

Reply:
The $request in process() and the request in the constructor are two different objects. The symphony request doesn't have the ability to log a user into concrete5, which is what the concrete5 request object offers here. That's the only reason it's being added and included. It's not ideal, obviously, and it's a prime candidate for deprecation in favor of something else in a later major release.

Reply:
Oh, interesting – apparently the actual object in the middleware is the Concrete request object? Hmm, perhaps this will be simpler then.

Reply:
N/m, I was wrong. Looks like I can just run the method on $request without injecting the other dependency. That's what I'll do.